packages/sds/src/message_channel/mem_local_history.ts

```diff
@@ -17,15 +19,71 @@ export const DEFAULT_MAX_LENGTH = 10_000;
  * If an array of items longer than `maxLength` is pushed, dropping will happen
  * at next push.
  */
-export class MemLocalHistory {
+export interface ILocalHistory {
+  length: number;
+  push(...items: ContentMessage[]): number;
+  some(
+    predicate: (
+      value: ContentMessage,
+      index: number,
+      array: ContentMessage[]
+    ) => unknown,
+    thisArg?: any
+  ): boolean;
+  slice(start?: number, end?: number): ContentMessage[];
+  find(
+    predicate: (
+      value: ContentMessage,
+      index: number,
+      obj: ContentMessage[]
+    ) => unknown,
+    thisArg?: any
+  ): ContentMessage | undefined;
+  findIndex(
+    predicate: (
+      value: ContentMessage,
+      index: number,
+      obj: ContentMessage[]
+    ) => unknown,
+    thisArg?: any
+  ): number;
+}
+
+export type MemLocalHistoryOptions = {
+  storage?: ChannelId | PersistentStorage;
```

Comment:
What are you doing here? Why would you use persistent storage for the Memory implementation?